doozer

```diff
@@ -1204,6 +1204,160 @@ def config_print(runtime, key, name_only):
+
+    The output SRC=DEST file is simply a new line separated file where
+    the LHS is a source image and the RHS is the destination to mirror
+    the image to. For example:
+
+    \b
+        registry.reg-aws.openshift.com:443/{repository}=quay.io/openshift-release-dev/ocp-v4.0-art-dev:{version}-{release}-{image_name_short}
+
+    Eample base ImageStream YAML:
+
+    \b
+        kind: ImageStream
+        apiVersion: image.openshift.io/v1
+        metadata:
+          name: 4.0-art-latest
+          namespace: ocp
+        spec:
+          tags: []
+
+    """
+    runtime.initialize(clone_distgits=False)
+    images = [i for i in runtime.image_metas()]
+
+    # Load the ImageStream stub file
+    with open(is_base) as fp:
+        isb = yaml.safe_load(fp)
+
+    # All the items we will put in the image mirror input
+    src_dest_items = []
+
+    for image in images:
+        dfp = DockerfileParser(path=runtime.working_dir)
+        try:
+            dfp.content = image.fetch_cgit_file("Dockerfile")
+        except Exception:
+            click.echo("Error reading Dockerfile from distgit: {}".format(image.distgit_key))
+            raise
+
+        version = dfp.labels["version"]
+
+        s = pattern
+        s = s.replace("{build}", "{component}-{version}-{release}")
+        s = s.replace("{repository}", "{image}:{version}-{release}")
+        s = s.replace("{namespace}", image.namespace)
+        s = s.replace("{name}", image.name)
+        s = s.replace("{image_name}", image.image_name)
+        s = s.replace("{image_name_short}", image.image_name_short)
+        s = s.replace("{component}", image.get_component_name())
+        s = s.replace("{image}", dfp.labels["name"])
+        s = s.replace("{version}", version)
+
+        release_query_needed = '{release}' in s or '{pushes}' in s
+
+        # Since querying release takes time, check before executing replace
+        release = ''
+        if release_query_needed:
+            try:
+                _, _, release = image.get_latest_build_info()
+            except IOError as err:
+                red_prefix("Error looking up build info (skipping): ")
+                click.echo(str(err))
+                continue
+
+        s = s.replace("{release}", release)
+
+        pushes_formatted = ''
+        for push_name in image.get_default_push_names():
+            pushes_formatted += '\t{} : [{}]\n'.format(push_name, ', '.join(image.get_default_push_tags(version, release)))
+
+        if pushes_formatted is '':
+            pushes_formatted = "(None)"
+
+        s = s.replace("{pushes}", '{}\n'.format(pushes_formatted))
+
+        if "{" in s:
+            raise IOError("Unrecognized fields remaining in pattern: %s" % s)
+
+        # Per clayton:
+        """Tim Bielawa: note to self: is only for `ose-` prefixed images
+        Clayton Coleman: Yes, Get with the naming system or get out of town
+        """
+        if 'ose' in image.image_name_short and '666' not in release:
```

Comment:
lol at 666. I'm sorry timmy. 
This should be something that eventually gets added to a config or passed in at the CLI. I will allow it now because I know this is experimental, but eventually of course we'll remove anything specific to us.